src/EoS/Nuclear/NuclearEoS.cpp

```diff
@@ -23,122 +40,213 @@ void findenergy( const real x, const real y, const real z,
+//                energy_shift    : energy_shift
+//                nrho            : size of density array in the Nuclear EoS table
+//                ntoreps         : size of (temperature/energy) array in the Nuclear EoS table (temp/energy-based table)
+//                nye             : size of Y_e array in the Nuclear EoS table
+//                nmode           : size of log(eps)   (0)
+//                                          entropy    (2)
+//                                          log(P)     (3) array in the Nuclear EoS table
+//                                                         for each mode
+//                alltables       : Nuclear EoS table
+//                alltables_mode  : Auxiliary log(T) arrays for energy mode
+//                                                              entropy mode
+//                                                              pressure mode
+//                logrho          : log(rho) array in the table
+//                logtoreps       : log(T) or log(eps) array for (T/eps) mode (temp/energy-based table)
+//                yes             : Y_e      array in the table
+//                logepsort_mode  : log(eps) or log(T) array for (eps/T) mode (temp/energy-based table)
+//                entropy_mode    : entropy  array for entropy mode
+//                logpress_mode   : log(P)   array for pressure mode
+//                keymode         : which mode we will use
+//                                  0 : energy mode      (coming in with eps)
+//                                  1 : temperature mode (coming in with T)
+//                                  2 : entropy mode     (coming in with entropy)
+//                                  3 : pressure mode    (coming in with P)
+//                keyerr          : output error
+//                                  667 : fail in finding T/e (temp/energy-based table)
+//                                  101 : Y_e too high
+//                                  102 : Y_e too low
+//                                  103 : temp too high 
+//                                  104 : temp too low
+//                                  105 : rho too high
+//                                  106 : rho too low
+//                                  107 : eps too high     (energy-based table)
+//                                  108 : eps too low      (energy-based table)
+//                                  109 : temp too high    (energy-based table)
+//                                  110 : temp too low     (energy-based table)
+//                                  111 : entropy too high (energy-based table)
+//                                  112 : entropy too low  (energy-based table)
+//                                  113 : log(P) too high  (energy-based table)
+//                                  114 : log(P) too low   (energy-based table)
+//                rfeps           : tolerence for interpolations
```

Comment:
- Why modify the structure, e.g. the (1) and (2) in line 34?
- Why change the leading characters from upper case to lower case?
- error code 107 - 114 would only occur in the energy-based table?
- some error code is missing here, e.g., 2xx, 66x